meta-balena-common/recipes-containers/balena/balena/balena-healthcheck

```diff
@@ -7,24 +7,46 @@ set -o errexit
 BALENAD_SOCKET="/run/balena.sock"
 CONTAINERD_SOCKET="/run/balena-engine/containerd/balena-engine-containerd.sock"
 
+# Checks if the file $1 contains an UUID in any of the formats described here:
+# https://github.com/google/uuid/blob/6e10cd1027e225e3ad7bfcc13c896abd165b02ef/uuid.go#L189
+contains_valid_uuid()
+{
+    # This matches xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx, which is the one format
+    # we have seen in practice, and is also the base for most other supported
+    # formats.
+    BASE_UUID_REGEX="[[:xdigit:]]{8}-([[:xdigit:]]{4}-){3}[[:xdigit:]]{12}"
+
+    # Check for all the possible formats, starting with the most common one, to
+    # benefit from the short-circuited evaluation. The contents of the uuid file
+    # must match these formats exactly, so we anchor the regexes both at the
+    # beginning (^) and the end ($).
```

Comment:
Love the detailed comments!

Reply:
I believe time travel is impossible, because otherwise I'd expect someone to have come back to thank me for those :stuck_out_tongue_closed_eyes: